### PR TITLE
Add actionlint

### PR DESCRIPTION
**Pull request by xmnlab**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xmnlab** [comment]
@jloehel this is an initial version of the recipe .. but it seems that it will need more work on that about the dependencies ... I will take a look into that later

**Review by jloehel — COMMENTED** [review]
Hi @xmnlab,

Thanks for opening the PR. Would be really nice to have it there. I am not super familiar with conda packaging. Maybe I'll ask some dummy questions. ^^ Where are the repos and architectures defined? How do you deal with conda on Microsoft Windows? Has conda some guidelines for go packaging? Do you include docs into the packages? Has the build service access to the internet to download something during the build?

File: recipes/actionlint/build.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+set -ex
+
+export GOPATH="${BUILD_PREFIX}/bin"
+export GOROOT="${BUILD_PREFIX}/go"
+
+make
+
+mkdir -p $PREFIX/bin
+
+mv out/minikube $PREFIX/bin
```
[the remark this answers is not in the document]
Comment (xmnlab) [reply recipes/actionlint/build.sh:12]: totally .. thanks for caching that :) as the `make` command is not working it didn't reach that line yet XD
I will fix that :) thanks

File: recipes/actionlint/build.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/env bash
+
+set -ex
+
+export GOPATH="${BUILD_PREFIX}/bin"
+export GOROOT="${BUILD_PREFIX}/go"
+
+make
```
[the remark this answers is not in the document]
Comment (xmnlab) [reply recipes/actionlint/build.sh:8]: thanks for pointing me to that. I will take a look into that.

File: recipes/actionlint/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  description: |
+    Syntax check for workflow files to check unexpected or missing keys following workflow syntax
+    Strong type check for `$\{\{ \}\}` expressions to catch several semantic errors like access to not existing property, type mismatches
+    Actions usage check to check that inputs at with: and outputs in steps.{id}.outputs are correct
+    shellcheck and pyflakes integrations for scripts at run:
+    Security checks; script injection by untrusted inputs, hard-coded credentials
+    Other several useful checks; glob syntax validation, dependencies check for needs:, runner label validation, cron syntax validation
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE.txt
```
[the remark this answers is not in the document]
Comment (xmnlab) [reply recipes/actionlint/meta.yaml:37]: yes. you can find more information here: https://conda-forge.org/docs/maintainer/adding_pkgs.html?highlight=packaging%20license#packaging-the-license-manually

**Comment by xmnlab** [comment]
@jloehel thanks for your questions, I will try to answer :)

> Where are the repos and architectures defined? 
if there is no filtering/selection it will be created for all architecture available. 
for example: https://conda-forge.org/docs/maintainer/adding_pkgs.html#build

> How do you deal with conda on Microsoft Windows?
inside meta.yaml you can define specific configuration for specific architecture using selectors, 
for example `# [win]` (see example above)
also you can create a specific build script for windows using `build.bat`

> Has conda some guidelines for go packaging?
not that I am aware

> Do you include docs into the packages?
AFAIK it is not necessary, but we have into a metadata file a link to the documentation page.

> Has the build service access to the internet to download something during the build?
yes, the common approach is to use everything from conda-forge, but in some cases it is necessary to define some system dependencies (not sure if it is still used). also there is a way to define some "rules" for the CI using conda-forge.yaml (https://conda-forge.org/docs/maintainer/adding_pkgs.html#conda-forge-yml).

File: recipes/actionlint/build.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jloehel) [reply recipes/actionlint/build.sh:12]: I used the more or less the build command from the official Dockerfile and not the make for the rpm:
```
%build
export DATE="$(date --iso-8601)"
export HASH="%{shortcommit}"
go build \
  -mod=vendor \
  -buildmode=pie \
  -ldflags "-w -X %{provider_prefix}.version=%{version} -X %{provider_prefix}.commit=${HASH} -X %{provider_prefix}.date=${DATE}" \
	-a -tags netgo -o execs/%{name} ./cmd/actionlint
```

**Comment by jloehel** [comment]
Thanks :-)

> > Has the build service access to the internet to download something during the build?
> 
> yes, the common approach is to use everything from conda-forge, but in some cases it is necessary to define some system dependencies (not sure if it is still used). also there is a way to define some "rules" for the CI using conda-forge.yaml (https://conda-forge.org/docs/maintainer/adding_pkgs.html#conda-forge-yml).

Great! Makes it much easier. ^^

**Pull request closed without merging**